src/Disks/StoragePolicy.cpp

```diff
@@ -210,6 +211,14 @@ ReservationPtr StoragePolicy::reserve(UInt64 bytes) const
 }
 
 
+ReservationPtr StoragePolicy::reserveAndCheck(UInt64 bytes) const
```

Comment:
Maybe get rid of `MergeTreeData::checkAndReturnReservation()`? Otherwise it may lead to similar problem, when using functions that required the mentioned method after their invocation.

Reply:
It's a good idea, we can rename `reserve(...)` to `tryReserve(...)` and `reserveAndCheck(...)` to just `reserve(...)`, so it will be consistent with other names. But we also have similar `reserve(UInt64 bytes)` method in `Space` (`IDisk`) interface and I don't want to change a lot of signatures in bugfix PR.